plugins/csp-wms-overlays/src/TextureOverlayRenderer.hpp

```diff
@@ -0,0 +1,141 @@
+#ifndef TEXTURE_OVERLAY_RENDERER
+#define TEXTURE_OVERLAY_RENDERER
+
+#include "Plugin.hpp"
+#include "WebMapLayer.hpp"
+#include "WebMapService.hpp"
+#include "WebMapTextureLoader.hpp"
+
+#include <VistaKernel/GraphicsManager/VistaOpenGLDraw.h>
+#include <VistaMath/VistaBoundingBox.h>
+#include <VistaOGLExt/VistaGLSLShader.h>
+
+#include <array>
+#include <functional>
+#include <future>
+#include <unordered_map>
+
+// FORWARD DEFINITIONS
+class VistaGLSLShader;
+class VistaOpenGLNode;
+class VistaTexture;
+class VistaViewport;
+
+namespace cs::core {
+class SolarSystem;
+class TimeControl;
+class Settings;
+} // namespace cs::core
+
+namespace csp::wmsoverlays {
+
+/// Class which gets a geo-referenced texture and overlays if onto the previous rendered scene.
+/// Therefore it copies the depth buffer first. Second, in the shader it does an inverse projection
+/// to get the cartesian coordinates. This coordinates are transformed to latitude and longitude to
+/// do the lookup in the geo-referenced texture. The value is then overlayed on that pixel position.
+class TextureOverlayRenderer : public IVistaOpenGLDraw {
+ public:
+  TextureOverlayRenderer(std::string center, std::shared_ptr<cs::core::SolarSystem> solarSystem,
+      std::shared_ptr<cs::core::TimeControl> timeControl,
+      std::shared_ptr<cs::core::Settings>    settings,
+      std::shared_ptr<Plugin::Settings>      pluginSettings);
+  virtual ~TextureOverlayRenderer();
+
+  /// Returns the SPICE name of the body to which this renderer is assigned.
+  std::string getCenter() const;
+
+  /// Configures the internal renderer according to the given values.
+  void configure(Plugin::Settings::Body const& settings);
```

Comment:
If `Plugin::Settings::Body` can be moved, I wouldn't use `const&` as this method copies the value. So the interface would be clearer without the ref.
```suggestion
  void configure(Plugin::Settings::Body settings);
```